rosbag2_cpp/src/rosbag2_cpp/message_definitions/local_message_definition_source.cpp

```diff
@@ -183,15 +193,27 @@ std::pair<Format, std::string> MessageDefinitionCache::get_full_text(
   std::string result;
   Format format = Format::MSG;
   try {
-    result = append_recursive(DefinitionIdentifier{format, root_package_resource_name});
+    result = append_recursive(DefinitionIdentifier(root_type_name, format));
```

Comment:
I don't really like unbounded recursion here.  
Recursion is evil in embedded systems. Especially when it tied to the input from file system.
Attacker can easily create cyclic link on message datatype and cause stackoverflow. 
Need to consider how to get rid from recursion here.

Reply:
Inserted a hard-check on recursion depth.